torch/nn/modules/_functions.py

```diff
@@ -49,7 +49,8 @@ def forward(self, input, weight, bias, running_mean, running_var, eps, momentum,
         # batch_norm_gather_stats_with_counts calculates global mean & invstd based on
         # all gathered mean, invstd and count.
         # for nccl backend, use the optimized version of all gather.
-        if process_group._get_backend_name() == 'nccl':
+        # The Gloo backend does not support `all_gather_into_tensor`.
+        if process_group._get_backend_name() != "gloo":
```

Comment:
cc @H-Huang does that look ok to you?

Reply:
@H-Huang @fegin  sorry to bother could you have a look?

Reply:
> This would fail for backends like MPI (or other 3rd party backends) are not gloo but which also dont support `all_gather_into_tensor`. Which backend are you targeting for this change and is there urgency behind this feature?
> 
> A better way would be to introduce collective fallbacks which try a collective on a backend and if it is not available then fallback to another collective. We want to move away from hard checks for backends in our code to be more backend agnostic. However this feature is not available and still involves discussion/design from the distributed team

yeah I will make a change,and we are custom device with privateuse1 backend, it is a cuda-like device, we also want to support syncBN and this optimize.